syntax = "proto3";

option java_multiple_files = true;
option java_package = "network.beneath.proto";
option java_outer_classname = "BeneathProto";

package proto;

import "engine.proto";

service Gateway {
  rpc WriteRecords(WriteRecordsRequest) returns (WriteRecordsResponse) {}
  rpc GetStreamDetails(StreamDetailsRequest) returns (StreamDetailsResponse) {}
}

message WriteRecordsResponse {
}

message StreamDetailsRequest {
  string project_name = 1;
  string stream_name = 2;
}

message StreamDetailsResponse {
  bytes current_instance_id = 1;
  bytes project_id = 2;
  bool public = 3;
  bool external = 4;
  bool batch = 5;
  bool manual = 6;
  repeated string key_fields = 7;
  string avro_schema = 8;
}
